src/utils.c

```diff
@@ -847,6 +848,7 @@ ts_sub_integer_from_now(int64 interval, Oid time_dim_type, Oid now_func)
 	Datum now;
 	int64 res;
 
+	Ensure(OidIsValid(now_func), "invalid now function for type %d", time_dim_type);
```

Comment:
This seems misplaced as the check only checks for valid Oid not whether the function actually matches what we need here.

Reply:
Removed.